cpp/src/join/quadtree_poly_filtering.cu

```diff
@@ -0,0 +1,292 @@
+                                                 cudaStream_t stream)
+  {
+    return join_quadtree_and_bboxes<T>(quadtree,
+                                       poly_bbox,
+                                       static_cast<T>(std::min(x_min, x_max)),
+                                       static_cast<T>(std::max(x_min, x_max)),
+                                       static_cast<T>(std::min(y_min, y_max)),
+                                       static_cast<T>(std::max(y_min, y_max)),
+                                       static_cast<T>(scale),
+                                       max_depth,
+                                       mr,
+                                       stream);
+  }
+  template <typename T,
+            std::enable_if_t<!std::is_floating_point<T>::value> * = nullptr,
+            typename... Args>
+  inline std::unique_ptr<cudf::table> operator()(Args &&...)
+  {
+    CUSPATIAL_FAIL("Only floating-point types are supported");
+  }
+};
+}  // namespace
+
+std::unique_ptr<cudf::table> quad_bbox_join(cudf::table_view const &quadtree,
+                                            cudf::table_view const &poly_bbox,
+                                            double x_min,
+                                            double x_max,
+                                            double y_min,
+                                            double y_max,
+                                            double scale,
+                                            cudf::size_type max_depth,
+                                            rmm::mr::device_memory_resource *mr,
+                                            cudaStream_t stream)
+{
+  return cudf::type_dispatcher(poly_bbox.column(0).type(),
+                               dispatch_quadtree_bounding_box_join{},
+                               quadtree,
+                               poly_bbox,
+                               x_min,
+                               x_max,
+                               y_min,
+                               y_max,
+                               scale,
+                               max_depth,
+                               mr,
+                               stream);
+}
+
+}  // namespace detail
+
+std::unique_ptr<cudf::table> quad_bbox_join(cudf::table_view const &quadtree,
+                                            cudf::table_view const &poly_bbox,
+                                            double x_min,
+                                            double x_max,
+                                            double y_min,
+                                            double y_max,
+                                            double scale,
+                                            cudf::size_type max_depth,
+                                            rmm::mr::device_memory_resource *mr)
+{
+  CUSPATIAL_EXPECTS(quadtree.num_columns() == 5, "quadtree table must have 5 columns");
```

Comment:
I didn't see tests for all of these expectations.

Reply:
Added a test for these :+1: